server/src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -509,7 +518,35 @@ private String createDN(Consumer consumer, Owner owner) {
     private byte[] createContentAccessDataPayload(Owner owner, Environment environment) throws IOException {
         // fake a product dto as a container for the org content
         Set<Product> containerSet = new HashSet<>();
-        CandlepinQuery<Product> ownerProduct = ownerProductCurator.getProductsByOwner(owner);
+        Set<String> productUuids = new HashSet<>();
+
+        Collection<String> activePoolIds = poolCurator.getActivePoolByOwnerId(owner.getId());
+        Map<String, Set<String>> products =
+            poolCurator.getProductAndDerivedProductUuidsByPoolIds(activePoolIds);
+
+        Map<String, Set<String>> providedProducts =
+            poolCurator.getProvidedProductByPoolIds(activePoolIds, false);
+
+        Map<String, Set<String>> derivedProvidedProducts =
+            poolCurator.getDerivedProvidedProductByPoolIds(activePoolIds, false);
+
+        for (String poolId : activePoolIds) {
+            if (providedProducts.get(poolId) != null) {
+                productUuids.addAll(providedProducts.get(poolId));
+            }
+
+            if (derivedProvidedProducts.get(poolId) != null) {
+                productUuids.addAll(derivedProvidedProducts.get(poolId));
+            }
+
+            if (products.get(poolId) != null) {
+                productUuids.addAll(products.get(poolId));
+            }
+        }
+
+        CandlepinQuery<Product> productsToConsider =
```

Comment:
CandlepinQuery is deprecated, we shouldn't introduce new code that uses it.

Reply:
Removed.